### ISSUE TITLE
 loading libcrypto in an unsafe way M1 MacOS

### ISSUE THREAD

**Issue by sahasvat** [philippe44/AirConnect#451]
WARNING: /Users/I554894/Downloads/folder/aircast-macos-arm64 is loading libcrypto in an unsafe way
zsh: abort      /Users/I554894/Downloads/folder/aircast-macos-arm64

can you help me resolve this issue??

using M1 Pro and latest MacOS

**Comment by philippe44** [comment]
See closed posts

**Comment by sahasvat** [comment]
https://github.com/philippe44/AirConnect/issues/445

Had a look on it.
please allow me to run with Static version

**Comment by sahasvat** [comment]
even if i run Static Version, I'm unable to find devices.. Why?

**Comment by philippe44** [comment]
Which Ethernet interface do you have and which one does it bound to? It must the same where your devices are. Post the log here

**Comment by sahasvat** [comment]
connected to wifi also have ensured that Speaker is connected to Wifi

The Logs:
Last login: Wed Mar  1 01:40:26 on ttys000
/Users/I554894/Downloads/folder/aircast-macos-arm64-static ; exit;              
I554894@PN2YP9K9W0 ~ % /Users/I554894/Downloads/folder/aircast-macos-arm64-static ; exit;
[01:40:35.151356] main:884 Starting aircast version: v1.0.14 (Jan 25 2023 @ 12:35:22)
[01:40:35.152501] main:891 no config file, using defaults
[01:40:35.152540] Start:662 Binding to 192.168.0.105



Spotify or Google Chrome is able to detect the Chromecast Speakers

**Comment by philippe44** [comment]
So you're 100% sure that 192.168.0.x is the network where the CC devices are?

**Comment by sahasvat** [comment]
yes

took this from my router logs



1 | Xiaomi | 28-D1-27-A7-D1-A7 | 192.168.0.100 | 15:42:17
2 | Xiaomi | 28-D1-27-AA-63-2B | 192.168.0.101 | 15:42:18
3 | W-C7SM1F3 | 48-51-C5-1D-53-12 | 192.168.0.102 | 16:22:48
4 | Unknown | BA-8A-41-7F-1D-9A | 192.168.0.103 | 20:57:08
5 | Unknown | 52-00-45-93-EE-9D | 192.168.0.104 | 20:01:25


all my CC devices are connected to 192.168.0.X

**Comment by eduardorangell** [comment]
Hi there, i was with the same issue (consequently the program did not list the devices on the network) and i was able to fix by creating 2 symlinks mentioned on [readme](https://github.com/philippe44/AirConnect#installing).

Basically i:
1. Installed openssl@3 through brew
2. created `libcrypto` and `libssl` dylib symlinks
3. restarted OS

i hope this can give you some light to fix in your computer.

**Comment by owlboy** [comment]
@eduardorangell The instructions in the readme are unclear for non-experts. I think I managed to create these symlinks. But even after a restart, I get the same "unsafe loading" error. 

The confusing part about the instructions in the readme is the part where the version number is optional. I entered `/3.1.0/` since that appeared to be the current correct number as of this post.  – Did you do something different?

Thanks @eduardorangell

**Comment by eduardorangell** [comment]
@owlboy Coincidently i had to format my macbook, so i needed to reinstall everything back (fresh install like you)

I noticed that this newer homebrew version changed where programs are installed, so this readme in fact could be a little confusing to that version specifically.

I did the same configuration like yours and after rebooting all worked fine.

- Had to create `/usr/local/lib` manually
- I ran `ln -s /opt/homebrew/opt/openssl@3.1/lib/libcrypto.dylib /usr/local/lib/libcrypto.dylib`
- I ran `ln -s /opt/homebrew/opt/openssl@3.1/lib/libssl.dylib /usr/local/lib/libssl.dylib`
- created plist mentioned on readme to run automatically

Try to `ls /opt/homebrew/opt` to check which openssl versions are installed and then set the path correctly when run symlink creation again.
